File: src/main/java/com/itsci/mju/maebanjumpen/partyrole/dto/HousekeeperDTO.java
package com.itsci.mju.maebanjumpen.partyrole.dto;

import com.itsci.mju.maebanjumpen.housekeeperskill.dto.HousekeeperSkillDTO;
import lombok.*;

import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class HousekeeperDTO extends MemberDTO {
    private String photoVerifyUrl;
    private String statusVerify;
    private Double rating;
    private String dailyRate;

    private List<Integer> hireIds;

    private Set<HousekeeperSkillDTO> housekeeperSkills;

    @Override
    public String getType() {
        return "housekeeper";
    }
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/dto/HousekeeperDTO.kt
package com.itsci.mju.maebanjumpen.partyrole.dto

import com.itsci.mju.maebanjumpen.housekeeperskill.dto.HousekeeperSkillDTO

open class HousekeeperDTO : MemberDTO() {
    var photoVerifyUrl: String? = null
    var statusVerify: String? = null
    var rating: Double? = null
    var dailyRate: String? = null
    var hireIds: List<Int>? = null
    var housekeeperSkills: Set<HousekeeperSkillDTO>? = null

    override fun getType(): String = "housekeeper"
}

